Objects/weakrefobject.c

```diff
@@ -46,38 +46,59 @@ new_weakref(PyObject *ob, PyObject *callback)
 }
 
 
-/* This function clears the passed-in reference and removes it from the
- * list of weak references for the referent.  This is the only code that
- * removes an item from the doubly-linked list of weak references for an
- * object; it is also responsible for clearing the callback slot.
+/* This function is responsible for clearing the callback slot. When using this
+ * function it must be called before calling remove_weakref_from_referent,
+ * otherwise a segmentation fault will occur at build time.
```

Comment:
The function should be clearing the weakref object.  So:

```suggestion
/* This function clears the passed-in reference.  It must have been removed
 * from the weaklist already.  (See weaklist_remove_weakref().)
```